internal/extproc/chatcompletion_processor.go

```diff
@@ -38,42 +38,52 @@ type chatCompletionProcessor struct {
 	costs translator.LLMTokenUsage
 }
 
+// selectTranslator selects the translator based on the output schema.
+func (c *chatCompletionProcessor) selectTranslator(out filterapi.VersionedAPISchema) error {
+	if c.translator != nil { // Prevents re-selection and allows translator injection in tests.
+		return nil
+	}
+	// TODO: currently, we ignore the LLMAPISchema."Version" field.
+	switch out.Name {
+	case filterapi.APISchemaOpenAI:
+		c.translator = translator.NewChatCompletionOpenAIToOpenAITranslator()
+	case filterapi.APISchemaAWSBedrock:
+		c.translator = translator.NewChatCompletionOpenAIToAWSBedrockTranslator()
+	default:
+		return fmt.Errorf("unsupported API schema: backend=%s", out)
+	}
+	return nil
+}
+
 // ProcessRequestHeaders implements [ProcessorIface.ProcessRequestHeaders].
-func (p *chatCompletionProcessor) ProcessRequestHeaders(_ context.Context, _ *corev3.HeaderMap) (res *extprocv3.ProcessingResponse, err error) {
+func (c *chatCompletionProcessor) ProcessRequestHeaders(_ context.Context, _ *corev3.HeaderMap) (res *extprocv3.ProcessingResponse, err error) {
 	// The request headers have already been at the time the processor was created
 	return &extprocv3.ProcessingResponse{Response: &extprocv3.ProcessingResponse_RequestHeaders{
 		RequestHeaders: &extprocv3.HeadersResponse{},
 	}}, nil
 }
 
 // ProcessRequestBody implements [ProcessorIface.ProcessRequestBody].
-func (p *chatCompletionProcessor) ProcessRequestBody(ctx context.Context, rawBody *extprocv3.HttpBody) (res *extprocv3.ProcessingResponse, err error) {
-	path := p.requestHeaders[":path"]
-	model, body, err := p.config.bodyParser(path, rawBody)
+func (c *chatCompletionProcessor) ProcessRequestBody(ctx context.Context, rawBody *extprocv3.HttpBody) (res *extprocv3.ProcessingResponse, err error) {
+	path := c.requestHeaders[":path"]
+	model, body, err := c.config.bodyParser(path, rawBody)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse request body: %w", err)
 	}
-	p.logger.Info("Processing request", "path", path, "model", model)
+	c.logger.Info("Processing request", "path", path, "model", model)
 
-	p.requestHeaders[p.config.modelNameHeaderKey] = model
-	b, err := p.config.router.Calculate(p.requestHeaders)
+	c.requestHeaders[c.config.modelNameHeaderKey] = model
+	b, err := c.config.router.Calculate(c.requestHeaders)
 	if err != nil {
 		return nil, fmt.Errorf("failed to calculate route: %w", err)
 	}
-	p.logger.Info("Selected backend", "backend", b.Name)
+	c.logger.Info("Selected backend", "backend", b.Name)
 
-	factory, ok := p.config.factories[b.Schema]
-	if !ok {
-		return nil, fmt.Errorf("failed to find factory for output schema %q", b.Schema)
-	}
-
-	t, err := factory(path)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create translator: %w", err)
+	if err = c.selectTranslator(b.Schema); err != nil {
```

Comment:
i feel like this could be called when calling NewProcessor instead of inside ProcessRequestBody? Then we won't need `.translator != nil` inside selectTranslator i guess?